src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -335,20 +335,19 @@ internal void LogBuffer(int size)
         private SocketError FinishOperationAccept(Internals.SocketAddress remoteSocketAddress)
         {
             System.Buffer.BlockCopy(_acceptBuffer!, 0, remoteSocketAddress.Buffer, 0, _acceptAddressBufferCount);
-            Socket? sukru = _acceptSocket;
+            Socket? tempAcceptSocket = _acceptSocket;
             _acceptSocket = _currentSocket!.CreateAcceptSocket(
                 SocketPal.CreateSocket(_acceptedFileDescriptor),
                 _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
-            if (sukru != null)
+            if (tempAcceptSocket != null)
             {
-                sukru.DisposeHandle();
-                sukru.CopyStateFromSource(_acceptSocket);
+                tempAcceptSocket.DisposeHandle();
+                tempAcceptSocket.CopyStateFromSource(_acceptSocket);
                 // We keep this socket to make clean-up.
                 Socket temp = _acceptSocket;
-                _acceptSocket = sukru;
+                _acceptSocket = tempAcceptSocket;
                 temp.ClearHandle();
                 temp.Dispose();
```

Comment:
I find the structure of this method confusing, in particular as it's taking the user-supplied socket out of the field, overwriting it with the accepted socket, and then replacing it again, with multiple temps along the way.

I'd find it clearer if the flow were instead:
```C#
Socket acceptedSocket = _currentSocket!.CreateAcceptSocket(
    SocketPal.CreateSocket(_acceptedFileDescriptor),
    _currentSocket._rightEndPoint!.Create(remoteSocketAddress));
if (_acceptSocket is null)
{
    // Store the accepted socket
    _acceptSocket = acceptedSocket;
}
else
{
    // Copy state from the accepted socket into the caller-supplied socket and then dispose of the original.
    _acceptSocket.DisposeHandle();
    _acceptSocket.CopyStateFromSource(acceptedSocket);
    acceptSocket.ClearHandle();
    acceptSocket.Dispose();
}
```

Reply:
Agreed, I will convert it in this way.